enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/codegen/GeneratedMethodStructure.scala

```diff
@@ -425,6 +425,12 @@ class GeneratedMethodStructure(val fields: Fields, val generator: CodeBlock, aux
 
   override def nodeGetRelationshipsWithDirection(iterVar: String, nodeVar: String, direction: SemanticDirection) = {
     val local = generator.declare(typeRef[RelationshipIterator], iterVar)
+    generator.assign(local, constant(null))
+    //Make sure cursor is always closed
+    _finalizers.append((_: Boolean) => (block) =>
+      using(block.ifStatement(Expression.notNull(block.load(local.name())))) {inner =>
+        inner.expression(invoke(inner.load(local.name()), method[RelationshipIterator, Unit]("close")))
+      })
```

Comment:
They might not be in scope at that point, the change I made will close the iterator on the exit of every loop (as well as on other early exits such as exceptions etc but that was already there). So the generated code will look something like

```java
RelationshipIterator it;
it = null;
try
{
     it = this.getOrLoadReadOperations().nodeGetRelationships( v1, org.neo4j.graphdb.Direction.OUTGOING );
 }
 catch ( org.neo4j.kernel.api.exceptions.KernelException e )
{
    if ( it != null )
    {
          it.close();
     }
     throw new org.neo4j.cypher.internal.frontend.v3_2.CypherExecutionException( e.getUserMessage( new org.neo4j.kernel.api.StatementTokenNameLookup( this.ro ) ), e );
}
 while( it.hasNext() )
{
...
  if ( it != null )
  {
     it.close();
  }
}
```
It will also be closed on all exceptions etc within that loop but it is too much typing